pkg/handlers/generic/mutation/extraapiservercertsans/inject.go

```diff
@@ -94,9 +106,17 @@ func (h *extraAPIServerCertSANsPatchHandler) Mutate(
 			if obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration == nil {
 				obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration = &bootstrapv1.ClusterConfiguration{}
 			}
-			obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.CertSANs = extraAPIServerCertSANsVar
-
+			obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.CertSANs = apiCertSANs
 			return nil
 		},
 	)
 }
+
+func getDefaultAPIServerSANs(cluster *clusterv1.Cluster) []string {
+	switch utils.GetProvider(cluster) {
+	case "docker":
+		return v1alpha1.DefaultDockerCertSANs
+	default:
+		return nil
```

Comment:
```suggestion
		return []string{}
```

Reply:
lol https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/519#discussion_r1567696950

lets just leave it unallocated

Reply:
sounds good.